tools/update_runnablejobs.py

```diff
@@ -1,98 +1,201 @@
+
+    This makes sure that we use a consistent key between our code and selecting jobs from the
+    table.
+
+    See row[1:4] usage in the code which selects the 2nd to 4th columns from jobpriority table.
+    '''
+    return (job['testtype'], job['platform_option'], job['platform'])
 
+
+def get_rootdir():
     path = os.path.expanduser('~/.mozilla/seta/')
     if not os.path.exists(path):
         os.makedirs(path)
 
     return path
 
 
-# TODO we need add test for this function to make sure it works under any situation
-def update_runnableapi():
-    """Use it to update runnablejobs.json file."""
+def sanitized_data(runnable_jobs_data):
+    '''We receive data from runnable jobs api and return the sanitized data that meets our needs.
+
+    This is a loop to remove duplicates (including buildsystem -> * transformations if needed)
+    By doing this, it allows us to have a single database query
+
+    It returns sanitized_list which will contain a subset which excludes:
+    * jobs that don't specify the build_platform
+    * jobs that don't specify the testtype
+    * if the job appears again, we replace build_system_type with '*'
+    '''
+    map = {}
+    sanitized_list = []
+    for job in runnable_jobs_data['results']:
+        # XXX: Once this code moves to TH, the database models will have foreign keys
+        #      to other tables. Specifically platform/build_platform and platform_option
+        if not valid_platform(job['build_platform']):
+            continue
+
+        testtype = parse_testtype(
+            build_system_type=job['build_system_type'],
+            job_type_name=job['job_type_name'],
+            platform_option=job['platform_option'],
+            refdata=job['ref_data_name'])
+
+        if not testtype:
+            continue
+
+        # NOTE: This is *all* the data we need from the runnable API
+        new_job = {
+            'platform': job['build_platform'],  # e.g. windows8-64
+            'testtype': testtype,  # e.g. web-platform-tests-1
+            'platform_option': job['platform_option'],  # e.g. {opt,debug}
+            'build_system_type': job['build_system_type']  # e.g. {buildbot,taskcluster,*}
+        }
+        # key = (platform, testtype, job['platform_option'])
+        key = _unique_key(new_job)
+
+        # Let's build a map of all the jobs and if duplicated change the build_system_type to *
+        if key not in map:
+            map[key] = job['build_system_type']
+            sanitized_list.append(new_job)
+
+        elif job['build_system_type'] != map[key]:
+            previous_job = copy.deepcopy(new_job)
+            previous_job['build_system_type'] = map[key]
+            previous_index = sanitized_list.index(previous_job)
+            # This will make so we *replace* the previous build system type with '*'
+            # This also guarantees that we don't have duplicates
+            sanitized_list[previous_index]['build_system_type'] = '*'
```

Comment:
No, I don't. All I really care about is creating a sanitized list with the minimum data we need.
map is only used internally in this function to help me determine which jobs we have seen before and update the build_system_type if seen twice.